interfaces/cython/cantera/composite.py

```diff
@@ -528,32 +528,47 @@ def __init__(self, phase, shape=(0,), states=None, extra=None, meta=None):
         reserved = self.__dir__()
 
         self._extra = OrderedDict()
+
+        if isinstance(extra, str):
+            extra = [extra]
+
         if isinstance(extra, dict):
             for name, v in extra.items():
                 if name in reserved:
                     raise ValueError(
                         "Unable to create extra column '{}': name is already "
                         "used by SolutionArray objects.".format(name))
                 if not np.shape(v):
-                    self._extra[name] = [v]*self._shape[0]
-                elif len(v) == self._shape[0]:
-                    self._extra[name] = list(v)
+                    self._extra[name] = np.array([v]).reshape(self._shape)
+                elif len(v) == self._shape or np.array(v).shape == self._shape:
+                    self._extra[name] = np.array(v)
                 else:
                     raise ValueError("Unable to map extra SolutionArray"
```

Comment:
I know you are not responsible for this line, but can you add a space after `SolutionArray` here to fix the formatting? Thanks :smile: